fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/PlayerPickupItemCallback.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.player;
+
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.ActionResult;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Callback for the player picking up an item entity.
+ * Is hooked in before the item is picked up.
+ *
+ * <p>Upon return:
+ * <ul><li>SUCCESS cancels further processing and picks up the item.
```

Comment:
note there is another `CONSUME` enum value.

Reply:
It doesn't seem like any of the current callbacks document or handle `CONSUME`. Is there an agreed-upon standard for what it should do?